wandb/sdk/launch/_project_spec.py

```diff
@@ -145,43 +145,68 @@ def get_entry_point(self, entry_point: str) -> "EntryPoint":
             return self._entry_points[entry_point]
         return self.add_entry_point(entry_point)
 
-    def _fetch_project_local(self, api: Api) -> None:
+    def _check_and_download_code_artifacts(self, internal_api: Api) -> bool:
+        public_api = wandb.PublicApi(
+            overrides={"base_url": internal_api.settings("base_url")}
+        )
+
+        run_entity, run_project, run_name = utils.parse_wandb_uri(self.uri)
+        run = public_api.run(f"{run_entity}/{run_project}/{run_name}")
+        run_artifacts = run.logged_artifacts()
+
+        for artifact in run_artifacts:
+            if hasattr(artifact, "type") and artifact.type == "code":
+                artifact.download(self.project_dir)
+                self.build_image = True
+                return True
+
+        return False
+
+    def _fetch_project_local(self, internal_api: Api) -> None:
         """Fetch a project into a local directory, returning the path to the local project directory."""
         assert self.source != LaunchSource.LOCAL
-        parsed_uri = self.uri
         if utils._is_wandb_uri(self.uri):
-            run_info = utils.fetch_wandb_project_run_info(self.uri, api)
-            if not run_info["git"]:
-                raise ExecutionError("Run must have git repo associated")
-            utils._fetch_git_repo(
-                self.project_dir, run_info["git"]["remote"], run_info["git"]["commit"]
-            )
-            patch = utils.fetch_project_diff(self.uri, api)
 
-            if patch:
-                utils.apply_patch(patch, self.project_dir)
+            run_info = utils.fetch_wandb_project_run_info(self.uri, internal_api)
             entry_point = run_info.get("codePath", run_info["program"])
-            # For cases where the entry point wasn't checked into git
-            if not os.path.exists(os.path.join(self.project_dir, entry_point)):
-                downloaded_entrypoint = utils.download_entry_point(
-                    self.uri, api, entry_point, self.project_dir
+
+            downloaded_code_artifact = self._check_and_download_code_artifacts(
+                internal_api
+            )
+
+            if not downloaded_code_artifact:
```

Comment:
we're guaranteed the code artifact is named the same thing as if we fetched the repo & also that there wouldn't be conflicts between the logged diff patch vs the code artifact right

Reply:
Code artifacts don't have a diff.patch associated with them. They are just the code.

Not sure what you mean by the "named the same thing as if we fetched the git repo". In the case of fetching the code artifact we don't bother with the git repo at all.

Reply:
just wanted to make sure that the locally downloaded code artifact is at the same file path as the downloaded git repo